sanity-io/vscode-sanity
# Support Sanity V3 and new groq defineQuery

This PR gives the extension some well deserved TLC:

* Upgrade the sanity client the package is using
* Upgrades to latest GROQ version to support newer features
* Upgrade vscode extension internal + engine requirements
* Support new `defineQuery` from the groq package
	* Note: We currently match on _all_ `defineQuery` usage, so if another package has a defineQuery implementation this will match it. I _think_ there's a way to first ensure that `defineQuery` is imported from `groq`, but couldn't get the grammar configuration and the RegExp to work.
* Load configuration from sanity.cli.ts, if multiple files are detected we ask the user.
	*  Should we remember the choice?
* Loads the parameters first from the local file, if it's not found, or if it's not complete, we ask the user to fill in the values
* Update the result view to also include the query, parameter and timings.

To test:

pull and checkout the branch, then run:

```
$ npm install
$ npm run vsce:package
$ code --install-extension  vscode-sanity-0.2.0.vsix
```

Then reload the vscode window
